CMakeLists.txt

```diff
@@ -1,6 +1,16 @@
 ﻿cmake_minimum_required(VERSION 3.18) # needed on win32
 
-project("purple-gowhatsapp" LANGUAGES C)
+set(PROJECT_NAME "purple-gowhatsapp" C)
+set(VERSION_MAJOR "1")
+set(VERSION_MINOR "12")
+set(VERSION_PATCH "0")
+set(VERSION "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
+
+# For now, we won't be doing any cross-compiling, so specifying amd64 here will cover most cases.
+# If you have a different processor and need to cross-compile, say, ARM, then this won't build the package you need until it's modified.
+set(ARCHITECTURE "amd64")

```

Comment:
What is the purpose of setting an this variable? I cannot find any documentation about it. Does it affect CMake's choice of platform? If so, builds on win32 would fail.

Reply:
This is just a variable i use here to package later.  The build will not break on windows or other architectures.

The solution I propose here is not very scalable.

Perhaps I re-work it to detect the OS that we need to support and package for?  Unfortunately this may bleed into the build .  It appears that CMAKE requires the details of pieces like CMAKE_SYSTEM_PROCESSOR to truly change the architecture.  More research required so I can support build/package behavior without this variable ..

Reply:
going to resolve this, I can see that we are in agreement for managing th is next to the packaging location.  no longer an issue.